src/GNDStk/gnds-tree-node.hpp

```diff
@@ -0,0 +1,444 @@
+
+   // meta_t
+   // forwards to meta(meta_t) above
+   template<class T>
+   decltype(auto) operator()(const meta_t<T> &kwd) const
+   {
+      return meta(kwd);
+   }
+
+   // child_t
+   // forwards to child(child_t) above
+   template<class T, class META, class CHILD>
+   decltype(auto) operator()(const child_t<T,META,CHILD> &kwd) const
+   {
+      return child(kwd);
+   }
+
+   // child_t, ...
+   // multi-argument
+   template<class T, class META, class CHILD, class... Ts>
+   decltype(auto) operator()(
+      const child_t<T,META,CHILD> &kwd,
+      Ts &&...ts
+   ) const {
+      return (*this)(kwd)(std::forward<Ts>(ts)...);
+   }
+
+}; // class Node
+
+
+
+// Node::static_found
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+bool Node<MCON,CCON>::static_found = false;
+
+
+
+// -----------------------------------------------------------------------------
+// write
+// -----------------------------------------------------------------------------
+
+// write(char *)
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+inline bool Node<MCON,CCON>::write(
+   const char * const file,
+   const int level
+) const {
+   // calls write(ostream) below
+   std::ofstream ofs(file);
+   return !write(ofs,level).fail();
+}
+
+
+// write(ostream)
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+std::ostream &Node<MCON,CCON>::write(
+   std::ostream &os,
+   const int level
+) const {
+   // indentation
+   const std::string icurr(indent* level   ,' ');
+   const std::string inext(indent*(level+1),' ');
+
+   // write name
+   os << icurr << name << ":" << std::endl;
+
+   // write metadata
+   for (const auto &meta : metadata)
+      os << inext << meta.first << ": " << meta.second << std::endl;
+
+   // write children
+   for (const auto &cptr : children)
+      if (cptr)
+         cptr->write(os,level+1);
+
+   // done
+   return os;
+}
+
+
+// operator<<
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+inline std::ostream &operator<<(std::ostream &os, const Node<MCON,CCON> &obj)
+{
+   // calls write(ostream) above
+   return obj.write(os);
+}
+
+
+
+// -----------------------------------------------------------------------------
+// normalize
+// -----------------------------------------------------------------------------
+
+namespace detail {
+
+// helper: strip
+// Strips leading digits and white-space from a string. Modifies
+// the original string, and also returns a reference to it.
+inline std::string &strip(std::string &name)
+{
+   int n = 0, ch; const int size = name.size();
+   while (n < size && (isdigit(ch=name[n]) || isspace(ch)))
+      n++;
+   return n ? (name = std::string(&name[n])) : name;
+}
+
+} // namespace detail
+
+
+
+// normalize
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+void Node<MCON,CCON>::normalize()
```

Comment:
Not sure what `normalize` does. Either explain or pick a more descriptive name.

Reply:
Answered somewhere else in all my comments. We can certainly change the name. I'll need to complete the thinking-through of all the JSON capabilities before knowing for sure if normalize() will stick around, be changed into something else or into several possible things, etc.